hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/hip632/IsAuthorizedTest.java

```diff
@@ -111,668 +101,895 @@ public class IsAuthorizedTest {
+                        .when(withOpContext((spec, opLog) -> {
+                            final var messageHash = new Digest().digest("submit".getBytes());
+
+                            final var edKey = spec.registry().getKey(ED25519_KEY);
+                            final var privateKey = spec.keys()
+                                    .getEd25519PrivateKey(
+                                            com.swirlds.common.utility.CommonUtils.hex(edKey.toByteArray())
+                                                    .substring(4));
+                            final var signedBytes = SignatureGenerator.signBytes(messageHash, privateKey);
+
+                            var call = contractCall(
+                                            HRC632_CONTRACT,
+                                            "isAuthorizedRawCall",
+                                            accountNum.get(),
+                                            messageHash,
+                                            signedBytes)
+                                    .via(recordName)
+                                    .gas(testCase.gasAmount());
+                            if (testCase.status() == INSUFFICIENT_GAS) {
+                                call = call.hasKnownStatus(CONTRACT_REVERT_EXECUTED);
+                            }
+                            allRunFor(spec, call);
+                        }));
+                final var hapiSpec = testCase.status() == SUCCESS
+                        ? throughWhen.then(getTxnRecord(recordName)
+                                .hasPriority(recordWith()
+                                        .status(SUCCESS)
+                                        .contractCallResult(resultWith().contractCallResult(BoolResult.flag(true)))))
+                        : throughWhen.then(childRecordsCheck(
+                                recordName,
+                                CONTRACT_REVERT_EXECUTED,
+                                recordWith()
+                                        .status(INSUFFICIENT_GAS)
+                                        .contractCallResult(resultWith()
+                                                .gasUsedIsInRange(
+                                                        (testCase.gasAmount()
+                                                                - GAS_BURNT_IN_ADDITION_TO_IS_AUTHORIZED_RAW_ALLOWANCE),
+                                                        testCase.gasAmount()))));
+                dynamicTests.add(hapiSpec);
+            }
+
+            return dynamicTests.stream().reduce(Stream::concat).get();
+        }
+    }
+
+    @Nested
+    @DisplayName("IsAuthorizedTests")
+    class IsAuthorizedTests {
+        @HapiTest
+        final Stream<DynamicTest> ecdsaHappyPath() {
+            return hapiTest(
+                    newKeyNamed(ECDSA_KEY).shape(SECP_256K1_SHAPE).generator(new RepeatableKeyGenerator()),
```

Comment:
whoever wrote this set of tests should probably figure out how to have some reusable methods

Reply:
I'll be replacing these tests with a parameterized test that runs through these use cases and many more (positive and negative).  These tests though show the way!

Reply:
To be fixed #16448